packages/x-data-grid/src/models/props/DataGridProps.ts

```diff
@@ -787,9 +787,10 @@ export interface DataGridPropsWithoutDefaultValue<R extends GridValidRowModel =
    * @template R
    * @param {R} newRow Row object with the new values.
    * @param {R} oldRow Row object with the old values.
+   * @param {GridRowId} rowId The id of the row.
    * @returns {Promise<R> | R} The final values to update the row.
    */
-  processRowUpdate?: (newRow: R, oldRow: R) => Promise<R> | R;
+  processRowUpdate?: (newRow: R, oldRow: R, rowId?: GridRowId) => Promise<R> | R;
```

Comment:
Why optional though? It seems that we can guarantee `rowId` presence.

Reply:
Yes we can 👍